src/main/java/io/github/moulberry/notenoughupdates/overlays/AuctionSearchOverlay.java

```diff
@@ -333,8 +362,54 @@ public static void close() {
 			}
 		}
 
-		if (!search.isEmpty()) NotEnoughUpdates.INSTANCE.sendChatMessage("/ahs " + search);
+		if (tes != null) {
+			if (search.length() <= 15) {
+				tes.signText[0] = new ChatComponentText(search.substring(0, Math.min(search.length(), 15)));
+			} else {
+				List<String> words = SPACE_SPLITTER.splitToList(search);
+
+				StringBuilder line0 = new StringBuilder();
+				StringBuilder line1 = new StringBuilder();
+
+				int currentLine = 0;
+				for (String word : words) {
+					if (currentLine == 0) {
+						if (line0.length() + word.length() > 15) {
+							currentLine++;
+						} else {
+							line0.append(word);
+							if (line0.length() >= 15) {
+								currentLine++;
+								continue;
+							} else {
+								line0.append(" ");
+							}
+						}
+					}
+					if (currentLine == 1) {
+						if (line1.length() + word.length() > 15) {
+							line1.append(word, 0, 15 - line1.length());
+							break;
+						} else {
+							line1.append(word);
+							if (line1.length() >= 15) {
+								break;
+							} else {
+								line1.append(" ");
+							}
+						}
+					}
+					if (line1.length() >= 15) break;
+				}
```

Comment:
```suggestion
				ListIterator<String> words = SPACE_SPLITTER.splitToList(search).listIterator();
				StringBuilder line0 = new StringBuilder();
				while (words.hasNext() ){
					String word = words.next();
					if (line0.length() + word.length() > 15) {
						words.previous();
						break;
					}
					line0.append(word).append(' ');
				}
				StringBuilder line1 = new StringBuilder();
				while (words.hasNext() ){
					String word = words.next();
					if (line1.length() + word.length() > 15) {
						break;
					}
					line1.append(word).append(' ');
				}
```

Reply:
This whole thing could also be a single regex: `(.{0,15})( (.{0,15}))?` or something along those lines